src/components/SkillNode.tsx

```diff
@@ -119,7 +121,8 @@ class SkillNode extends React.Component<Props, State> {
     const parentNodeIsSelected =
       !parentNodeId || parentState === SELECTED_STATE;
 
-    if (nodeState === UNLOCKED_STATE && !parentNodeIsSelected) {
+    if (nodeState === SELECTED_STATE && !parentNodeIsSelected) {
+      this.context.decrementSelectedSkillCount();
```

Comment:
I've only tested quickly, but you still need to check whether the state is unlocked when it updated. If you replace the code with this, add some tests then it should be fine!

```
    if (nodeState === SELECTED_STATE && !parentNodeIsSelected) {
      this.context.decrementSelectedSkillCount();
      return this.updateState(LOCKED_STATE);
    }

    if (nodeState === UNLOCKED_STATE && !parentNodeIsSelected) {
      return this.updateState(LOCKED_STATE);
    }

```